### PR TITLE
Device groups: Add devices.

### PR DESCRIPTION
**Pull request by ldjebran**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # THEEDGE-1594)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `go fmt ./...` to check that my code is properly formatted
- [X] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

File: pkg/services/devicegroups.go [hunk 1]
```diff
@@ -108,3 +111,59 @@ func (s *DeviceGroupsService) GetDeviceGroupByID(ID string) (*models.DeviceGroup
 	}
 	return &deviceGroup, nil
 }
+
+const (
```
Comment (acosferreira) [inline pkg/services/devicegroups.go:115]: What do you think to move it for the Error class?

File: pkg/routes/devicegroups.go [hunk 2]
```diff
@@ -22,6 +22,19 @@ import (
 type deviceGroupTypeKey int
 
 const deviceGroupKey deviceGroupTypeKey = iota
+const contextDeviceGroupNotFound = "context device group not found"
+
+func setContextDeviceGroup(ctx context.Context, deviceGroup *models.DeviceGroup) context.Context {
+	return context.WithValue(ctx, deviceGroupKey, deviceGroup)
+}
+
+func getContextDeviceGroup(ctx context.Context) (*models.DeviceGroup, error) {
```
Comment (acosferreira) [inline pkg/routes/devicegroups.go:31]: I think we are doing this duplicated, https://github.com/RedHatInsights/edge-api/pull/620/files#diff-48f6502941a54bb08d271267b7002cd524e07fade2ffb11ba7437baca89f781dR253
Maybe can just get it once. 
What do you think?

File: pkg/routes/devicegroups.go [hunk 3]
```diff
@@ -22,6 +22,19 @@ import (
 type deviceGroupTypeKey int
 
 const deviceGroupKey deviceGroupTypeKey = iota
+const contextDeviceGroupNotFound = "context device group not found"
+
+func setContextDeviceGroup(ctx context.Context, deviceGroup *models.DeviceGroup) context.Context {
+	return context.WithValue(ctx, deviceGroupKey, deviceGroup)
```
Comment (acosferreira) [inline pkg/routes/devicegroups.go:28]: we are already setting the context in the route, should we set it again?
https://github.com/RedHatInsights/edge-api/pull/620/files#diff-48f6502941a54bb08d271267b7002cd524e07fade2ffb11ba7437baca89f781dR44

File: pkg/routes/devicegroups.go (hunk 2 above)
Comment (ldjebran) [reply pkg/routes/devicegroups.go:31]: cannot get it where we are duplicating this  ?

Comment (acosferreira) [reply pkg/routes/devicegroups.go:31]: https://github.com/RedHatInsights/edge-api/blob/main/pkg/routes/devicegroups.go#L224

File: pkg/routes/devicegroups.go (hunk 3 above)
Comment (acosferreira) [reply pkg/routes/devicegroups.go:28]: https://github.com/RedHatInsights/edge-api/blob/main/pkg/routes/devicegroups.go#L31

File: pkg/routes/devicegroups.go (hunk 2 above)
Comment (ldjebran) [reply pkg/routes/devicegroups.go:31]: Think the name getContextDeviceGoup is more adequate to the usage what do you think ?

Comment (acosferreira) [reply pkg/routes/devicegroups.go:31]: I think we can change the name, but avoid the duplicity ;)

File: pkg/routes/devicegroups.go (hunk 3 above)
Comment (acosferreira) [reply pkg/routes/devicegroups.go:28]: @biancarosa any thougth ?

Comment (biancarosa) [reply pkg/routes/devicegroups.go:27]: looks like the call was just encapsulated to be called on a test, nothing wrong with it

**Review by acosferreira — APPROVED** [review]
lgtm

**Pull request merged**